src/Build/Utilities/FowlerNollVo1aHash.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Utilities
+{
+    internal static class FowlerNollVo1aHash
```

Comment:
The [StringTools library](https://github.com/dotnet/msbuild/tree/main/src/StringTools) would be a slightly preferred place for this.